### PR TITLE
Fix to GNIRS XD 32 wavelengths for bluest order

### PR DESCRIPTION
**Pull request by jhennawi**
Fixed an issue with continuum subtraction and threshold determination for arc line spectra that have significant zero-padding. The arc line peak threshold was being set to be too high.

We probably need to run the full test suites and I could use some help with that since I'm not yet running in the cloud.

### CODE REVIEW

**Review by feigewang — APPROVED** [review]
I think running test is necessary for such change!

File: pypeit/core/arc.py [hunk 1]
```diff
@@ -879,7 +879,7 @@ def detect_lines(censpec, sigdetect=5.0, fwhm=4.0, fit_frac_fwhm=1.25, input_thr
 
     bpm: `numpy.ndarray`_, optional
         Bad-pixel mask for input spectrum. If None, all pixels
-        considered good.
+        considered good. If passed in shape must match that of censpec
```
Comment (kbwestfall) [inline pypeit/core/arc.py:882]: Thanks for the doc improvements!

File: pypeit/core/arc.py [hunk 2]
```diff
@@ -916,25 +916,27 @@ def detect_lines(censpec, sigdetect=5.0, fwhm=4.0, fit_frac_fwhm=1.25, input_thr
...
-    xrng = np.arange(detns.size, dtype=np.float)
+    # OLD CODE
+    #if len(censpec.shape) == 3:
+    #    detns = censpec[:, 0].flatten()
+    #else:
+    #    detns = censpec.copy()
+    #detns = detns.astype(np.float)
+
+    bpm_out = censpec == 0.0 if bpm is None else bpm
+    xrng = np.arange(censpec.size, dtype=np.float)
```
Comment (kbwestfall) [inline pypeit/core/arc.py:929]: Is there a case where the calling function passes a 3D array?  If this works, I'd suggest removing the commented code and the TODO comment above.

**Review by kbwestfall — COMMENTED** [review]
Looks good to me.  I'm only leaving a "comment" for now until we've checked that tests and the dev-suite pass.

**Comment by kbwestfall** [comment]
```======================= 324 passed in 1421.55s (0:23:41) =======================```

**Review by kbwestfall — APPROVED** [review]

**Pull request merged**